core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -637,6 +637,16 @@ private[deploy] class Master(
       }
       freeWorkers = freeWorkers.filter(canLaunchExecutor)
     }
+
+    val numExecutorsScheduled = assignedExecutors.sum
+    val numExecutorsLaunched = app.executors.size
+    // Check to see if we managed to launch the requested number of executors
+    if(numUsable != 0 && numExecutorsLaunched != app.executorLimit &&
+      numExecutorsScheduled != app.executorLimit) {
```

Comment:
Another thing is, how noisy is this? Do we log this if dynamic allocation is turned on (we shouldn't)?